desc/objectives/_free_boundary.py

```diff
@@ -0,0 +1,945 @@
+                sym=False,
+            )
+        else:
+            source_grid = self._source_grid
+
+        if self._eval_grid is None:
+            eval_grid = LinearGrid(
+                rho=np.array([1.0]),
+                M=eq.M_grid,
+                N=eq.N_grid,
+                NFP=int(eq.NFP),
+                sym=False,
+            )
+        else:
+            eval_grid = self._eval_grid
+
+        errorif(
+            not np.all(source_grid.nodes[:, 0] == 1.0),
+            ValueError,
+            "source_grid contains nodes not on rho=1",
+        )
+        errorif(
+            not np.all(eval_grid.nodes[:, 0] == 1.0),
+            ValueError,
+            "eval_grid contains nodes not on rho=1",
+        )
+        errorif(
+            source_grid.sym,
+            ValueError,
+            "Source grids for singular integrals must be non-symmetric",
+        )
+        if self._s is None:
+            k = min(source_grid.num_theta, source_grid.num_zeta * source_grid.NFP)
+            self._s = k - 1
+        if self._q is None:
+            k = min(source_grid.num_theta, source_grid.num_zeta * source_grid.NFP)
+            self._q = k // 2 + int(np.sqrt(k))
+
+        try:
+            interpolator = FFTInterpolator(eval_grid, source_grid, self._s, self._q)
+        except AssertionError as e:
+            warnings.warn(
+                "Could not built fft interpolator, switching to dft method which is"
+                " much slower. Reason: " + str(e)
+            )
+            interpolator = DFTInterpolator(eval_grid, source_grid, self._s, self._q)
+
+        edge_pres = np.max(np.abs(eq.compute("p", grid=eval_grid)["p"]))
+        warnif(
+            (edge_pres * mu_0 > 1e-6) and not self._sheet_current,
+            UserWarning,
+            f"Boundary pressure is nonzero (max {edge_pres} Pa), "
+            + "a sheet current should be included.",
+        )
+
+        self._eq_data_keys = [
+            "K_vc",
+            "B",
+            "|B|^2",
+            "R",
+            "phi",
+            "Z",
+            "e^rho",
+            "n_rho",
+            "|e_theta x e_zeta|",
+            "p",
+        ]
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        src_profiles = get_profiles(self._eq_data_keys, obj=eq, grid=source_grid)
+        src_transforms = get_transforms(self._eq_data_keys, obj=eq, grid=source_grid)
+        eval_profiles = get_profiles(self._eq_data_keys, obj=eq, grid=eval_grid)
+        eval_transforms = get_transforms(self._eq_data_keys, obj=eq, grid=eval_grid)
+
+        neq = 3 if self._sheet_current else 2  # number of equations we're using
+
+        self._constants = {
+            "eval_transforms": eval_transforms,
+            "eval_profiles": eval_profiles,
+            "src_transforms": src_transforms,
+            "src_profiles": src_profiles,
```

Comment:
See other comment about consistently using `source_` instead of `src_`. But I'm being a little picky here, since these are internal variables that the user shouldn't be interacting with.